src/ui.rs

```diff
@@ -239,9 +239,9 @@ enum UiOutputPair<'output> {
 }
 
 /// Wrapper to implement `Write` for locked `Box<dyn Write>`.
-struct DynWriteLock<'a, 'output>(MutexGuard<'a, Box<dyn Write + 'output>>);
+struct DynWriteLock<'a, T: Write>(MutexGuard<'a, T>);
```

Comment:
Is the `: Write` needed here? I recently wondered what the tradeoffs are between putting trait bounds on the struct or only on the impls. I then found https://stackoverflow.com/questions/49229332/should-trait-bounds-be-duplicated-in-struct-and-impl which recommends putting them only on the impls. What do you think?

Reply:
Good catch. I agree with that guidance, was just not thinking very hard about this mechanical transformation.